### ISSUE TITLE
Sidebar scrolling

### ISSUE THREAD

**Issue by edschofield** [ryan-roemer/sphinx-bootstrap-theme#136]
On this page (http://python-future.org/compatible_idioms.html) it's impossible to scroll the sidebar, which contains the table of contents for the page. It of course would be much easier to navigate a page such as this if the table of contents in the sidebar were scrollable. Even though the sidebar is long, the actual page is much longer!

It seems that the problem is that the ``affix`` class is added to the #sidebar ``div``. This isn't present in the HTML sources, so it must be added dynamically in JS.

Is it possible to disable this?

BTW: I'm surprised that nobody seems to have reported this issue before. Is my site the only one with a long sidebar?!

**Comment by ryan-roemer** [comment]
Confirmed.

<img width="1270" alt="screen shot 2015-08-03 at 1 09 47 pm" src="https://cloud.githubusercontent.com/assets/445557/9046665/137b457a-39e1-11e5-8f00-9830532c00cc.png">

Options:

* Disable the affix via config (?)
* Calculate when sidebar hits bottom and scroll up...

**Comment by ryan-roemer** [comment]
@edschofield -- I suspect this code (by me) broke things around https://github.com/ryan-roemer/sphinx-bootstrap-theme/commit/e44c82ab1b46fca2f8ccc01e391b7999b9f6f12e#diff-a9a917ff3002ba773a6dc158772e8b2cL97 when I removed the `bottom` parameter to the affix call.

I'll try and get some time later to experiment, but if you get a chance and want to throw in a `bottom` calculation into a PR, that would be awesome!

**Comment by tdegeus** [comment]
I have the same wish! It would be great if it would be fixed. Unfortunately my programming talents are more with algebra, so I cannot directly contribute.

**Comment by edschofield** [comment]
@ryan-roemer : thanks so much for working on this!

I have tested your PR #137 with a local version of http://python-future.org/compatible_idioms.html built against your new ``bug-sidenav-affix`` branch and it doesn't fix the issue for me. I have tried Safari and Chrome on OS X and the behaviour with both browsers is still that the sidebar scrolls for a few pixels and then snaps back to a fixed position while the rest of the page is scrollable.

I apologise that my JavaScript foo isn't up to scratch to be able to contribute to this, other than further testing!

**Comment by ryan-roemer** [comment]
@edschofield -- Can you try again? I pulled your repo and built it, and it _does_ look fixed to me:

Page top

<img width="1464" alt="screen shot 2015-10-14 at 5 12 01 pm" src="https://cloud.githubusercontent.com/assets/445557/10501468/c4ca2b8c-7296-11e5-9848-f8212e7aa75a.png">

Page bottom

<img width="1464" alt="screen shot 2015-10-14 at 5 12 04 pm" src="https://cloud.githubusercontent.com/assets/445557/10501469/cabb816c-7296-11e5-8f7a-71e9b31fe6dd.png">

**Comment by ryan-roemer** [comment]
And for reference, when I just use the existing PyPi version of `sphinx-bootstrap-theme`, I **do** encounter the bug that you opened the issue around.

Maybe a full clean / rebuild and bunch of browser refreshes? ;)

**Comment by ryan-roemer** [comment]
Released in https://pypi.python.org/pypi/sphinx-bootstrap-theme/0.4.8

**Comment by edschofield** [comment]
@ryan-roemer : thanks for working on this and apologies for my delayed reply...

I have tried it again with v0.4.8 and the behaviour I see is this: the scrolling of the sidebar works when scrolling up from the bottom of the page, but the sidebar doesn't scroll until then. In the case of the page I linked before (http://python-future.org/compatible_idioms.html) this means that the table of contents in the sidebar doesn't scroll to reveal further links unless someone thinks to scroll to the end of the entire page. I expect that, for the vast majority of users, table-of-contents links below the first screenful in the sidebar will still appear inaccessible, because the sidebar does not scroll as they start to scroll though the main page. For long pages like this one, they may never reach the end and therefore not realise that the sidebar is scrollable eventually.

I would like a feature to disable the sidebar affixing so that the sidebar scrolls just like the rest of the page. As you mentioned initially, this could perhaps be a per-page config option, or (if possible to implement) it could perhaps be the default behaviour whenever the sidebar contents are longer than one screenful.

Does this make sense?
